gateway/ln-gateway/src/actor.rs

```diff
@@ -149,6 +151,10 @@ impl GatewayActor {
         gw_rpc: GatewayRpcSender,
     ) -> Result<Self> {
         let register_client = client.clone();
+        let registration = register_client
+            .config()
+            .to_gateway_registration_info(route_hints.clone(), GW_ANNOUNCEMENT_TTL);
+        let registration_sent = registration.clone();
         task_group
             .spawn("Register with federation", |_| async move {
```

Comment:
Nit: this code always confused me, probably should return to this sometime and fix it. I don't understand why it is safe to do this in a separate thread and continue, even if the registration with the federation fails.

Reply:
Oh I remember, it needs to periodically register with the federation. We should probably move this to a new function to be clearer. Not blocking on this, just thinking out loud

Reply:
Yeah might want to change the retry logic, possibly block here.